drivers/flash/flash_stm32_generic.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2017 BayLibre, SAS
+ * Copyright (c) 2019 Linaro Limited
+ * Copyright (c) 2020 Andreas Sandberg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_stm32generic, CONFIG_FLASH_LOG_LEVEL);
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include "flash_stm32.h"
+
+#define WRITE_BLOCK_SIZE DT_PROP(DT_INST(0, soc_nv_flash), write_block_size)
+
+#if WRITE_BLOCK_SIZE == 8
+typedef uint64_t flash_prg_t;
+#elif WRITE_BLOCK_SIZE == 4
+typedef uint32_t flash_prg_t;
+#elif WRITE_BLOCK_SIZE == 2
+typedef uint16_t flash_prg_t;
+#else
+#error Unknown write block size
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32L0X)
+#define FLASH_ERASED_VALUE 0
+#else
+#define FLASH_ERASED_VALUE ((flash_prg_t)-1)
+#endif
```

Comment:
Similar situation as with `WRITE_BLOCK_SIZE`

Reply:
> Similar situation as with `WRITE_BLOCK_SIZE`

I have updated `flash_stm32.c` to reflect this.